api/v1alpha1/clusterextension_types.go

```diff
@@ -73,8 +73,37 @@ type ClusterExtensionSpec struct {
 	// the bundle may contain resources that are cluster-scoped or that are
 	// installed in a different namespace. This namespace is expected to exist.
 	InstallNamespace string `json:"installNamespace"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be applied or skipped when attempting to install the cluster extension.
+	Preflight PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is a custom struct that holds the necessary configuration values to
+// enable or disable the CRD Upgrade Safety validations. Currently, this holds CRDUpgradeSafetyPreflightConfigType
+// that sets the CRD Upgrade Safety validations to either Enabled/Disabled.
+type CRDUpgradeSafetyPreflightConfig struct {
+	CRDUpgradeSafetyPreflightConfigType CRDUpgradeSafetyPreflightConfigType `json:"crdUpgradeSafetyPreflightConfigType,omitempty"`
```

Comment:
This name feels very long. Maybe something like `mode` would be nicer to type?

Reply:
I think we should also add a default here and have it be set to `Enabled`

Reply:
Can use https://github.com/operator-framework/operator-controller/pull/845/files#diff-8e9c0eb955489f1a05a95a75a25842a010fdc02029ce06aae7d2163b895c17cbR62-R64 as inspiration for validations and defaults

Reply:
Yeah, updated it to `Mode` and set the default to `Enabled`

https://github.com/operator-framework/operator-controller/pull/845/commits/c4dd90710fb62661560e8972ae87aea92f8f76c6